exercises/slot_machine.livemd

```diff
@@ -46,46 +49,91 @@ defmodule SlotMachine do
   Make sure that the machine doesn't give more rewards than its balance.
   Make sure the machine doesn't give negative rewards!
 
-  Use the :rand erlang module to generate random numbers.
-
-  The machine should either print "You lose!" or "You won ... dollars"
+  The machine should either print "You lose!" or "You won ... pennies"
   """
   def roll(pid) do
+    chance = Enum.random([1, 2, 3])
+
+    # Only give reward 1 in 3 times 
+    # Don't give reward if the machine does not have any money in it
+    if chance != 1 || get_current_balance(pid) == 0 do
+      IO.puts("You lose!")
+    else
+      # Provide random reward between 1 and 7000 pennies
+      reward = Enum.random(1..7_000)
+
+      # Don't give reward higher tham machine balance
+      if reward > get_current_balance(pid) do
```

Comment:
Nested-if statements are usually an indication we can refactor our code. They are very difficult to read, especially if they include else clauses.

Here are a few ways to simplify it:
1. Use the current balance to calculate the random reward amount
2. Use a different control flow option such as `cond`

```elixir
winner = Enum.random([1, 2, 3]) <= 2
balance = get_current_balance(pid)
reward = Enum.random(1..balance)
empty = balance == 0 

message = cond do
  # it might be nice to have a different message for an empty machine.
  empty  -> "You lose!"
  not winner -> "You lose!"
  winner -> "You won #{reward} pennies!"
end
```